arkane/encorr/mbac.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#                                                                             #
+# RMG - Reaction Mechanism Generator                                          #
+#                                                                             #
+# Copyright (c) 2002-2019 Prof. William H. Green (whgreen@mit.edu),           #
+# Prof. Richard H. West (r.west@neu.edu) and the RMG Team (rmg_dev@mit.edu)   #
+#                                                                             #
+# Permission is hereby granted, free of charge, to any person obtaining a     #
+# copy of this software and associated documentation files (the 'Software'),  #
+# to deal in the Software without restriction, including without limitation   #
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,    #
+# and/or sell copies of the Software, and to permit persons to whom the       #
+# Software is furnished to do so, subject to the following conditions:        #
+#                                                                             #
+# The above copyright notice and this permission notice shall be included in  #
+# all copies or substantial portions of the Software.                         #
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This module provides methods for applying Melius-type bond additivity
+corrections (M-BAC) as described in:
+Anantharaman and Melius, J. Phys. Chem. A 2005, 109, 1734-1747
+"""
+
+import numpy as np
+import pybel
+
+from rmgpy.molecule import Molecule, Atom, Bond, getElement
+
+import arkane.encorr.data as data
+
+atom_spins = {
+    'H': 0.5, 'C': 1.0, 'N': 1.5, 'O': 1.0, 'F': 0.5, 'Si': 1.0, 'P': 1.5, 'S': 1.0, 'Cl': 0.5, 'Br': 0.5, 'I': 0.5
+}
+
+
+def get_bac_correction(model_chemistry, coords, nums, multiplicity=1, mol_corr=0.0):
+    """
+    Given the model chemistry, molecular coordinates, atomic numbers,
+    and dictionaries of BAC parameters, return the total BAC correction
+    (should be SUBTRACTED from energy).
+
+    Note that a molecular correction term other than 0 destroys the size
```

Comment:
Could you explain this point? What's a molecular correction term? Is it a general name for BACs and AECs? What are the practical implications of destroying the QC method consistency?

Reply:
Equation 3 and 4 in the Melius paper describe the molecular correction term. Basically, it is a single additional parameter that corrects for potential systematic errors in the overall electronic structure of the molecule due to the difference in spin quantum number of the molecule and summed spin quantum numbers of the atoms. The paper gives a good example of what it means to destroy size consistency: "For example, the sum of the calculated energies for hydrogen and oxygen atoms is not the same as the energy of the hydroxyl radical when the two atoms are fixed at infinity." Size consistency is a desirable property of quantum chemistry methods, but I don't actually know how much of an issue it would be in practice to have this term. So far, I haven't observed any benefit when fitting the term in addition to the other parameters so right now I'm just passing it as an argument to the function with a value of 0 and don't plan on including it in the `mbac` dictionary in `data.py`.